CYCLONEDDS_QUALITY_DECLARATION.md

```diff
@@ -100,27 +125,42 @@ In all, the test code appears to comprise approximately 25% of the codebase.
 
 ### Feature Testing [4.i]
 
+Each feature in `CycloneDDS` has corresponding tests which simulate typical usage, and they are located in separate directories next to the sources. New features are required to have tests before being added.
+Currently nightly results can be seen here:
+* [linux-aarch64_release](https://ci.ros2.org/view/nightly/job/nightly_linux-aarch64_release/lastBuild/testReport/rmw_cyclonedds_cpp/)
+* [linux_release](https://ci.ros2.org/view/nightly/job/nightly_linux_release/lastBuild/testReport/rmw_cyclonedds_cpp/)
+* [mac_osx_release](https://ci.ros2.org/view/nightly/job/nightly_osx_release/lastBuild/testReport/rmw_cyclonedds_cpp/)
+* [windows_release](https://ci.ros2.org/view/nightly/job/nightly_win_rel/lastBuild/testReport/rmw_cyclonedds_cpp/)
+
 A substantial amount of the tests found throughout the source tree appear to verify functionality of various features of `CycloneDDS`.
 However the lack of a complete feature list (see section [3.i]) makes it difficult to analyze the breadth of the tests.
 
 ### Public API Testing [4.ii]
 
+Each part of the public API has tests, and new additions or changes to the public API require tests before being added. The tests aim to cover both typical usage and corner cases, but are quantified by contributing to code coverage.
 There are some tests throughout the `CycloneDDS` source tree which specifically target the public API, but there are no policies or mechanisms to discover and measure which tests target the public API or how much of the public API is covered by the existing tests.
 
 ### Coverage [4.iii]
 
-There is no test coverage tracking in `CycloneDDS`.
+There is no test coverage tracking in `CycloneDDS`. Automated coverage runs are being added to the CI to show coverage.
+
 
 ### Performance [4.iv]
 
+`CycloneDDS` performance is verifiably good and regression free per ROS Tooling WG's [nightly CI performance tests](http://build.ros2.org/job/Fci__nightly-performance_ubuntu_focal_amd64/ ).
 Though the `CycloneDDS` [documentation](https://github.com/eclipse-cyclonedds/cyclonedds#performance) discusses the product's performance and discusses what metrics describe it, very few tests seem to validate the given statistics or ensure that the performance does not regress.
 One of the [example projects](https://github.com/eclipse-cyclonedds/cyclonedds/blob/15e68152c9d14105e87ab1afc7e5af9c9589f776/examples/throughput/readme.rst) can be used to measure the throughput of the product, but does not provide a mechanism for analyzing resource usage or latency.
```

Comment:
This should no longer be referenced, the tool to use is [ddsperf](https://github.com/eclipse-cyclonedds/cyclonedds/tree/master/src/tools/ddsperf).

While there are no automated, public tests or results (more due to a lack of reliable infrastructure than to policy), there is evidence in PRs that it does get taken into account (see, e.g., https://github.com/eclipse-cyclonedds/cyclonedds/pull/558). Finally, `ddsperf` is used to check for performance regressions regularly and before releases.

Reply:
updated per @eboasson